### PR TITLE
Added possibility to provide static SAPI config

### PR DESCRIPTION
**Pull request by votsa**
## Description

Added possibility to provide static SAPI config 

## Clubhouse Story

https://app.clubhouse.io/findhotel/story/31073/launch-sapi-a-b-test-with-a-50-50-split

[ch31073]

### CODE REVIEW

File: packages/core/src/configs/app-config.ts [hunk 1]
```diff
@@ -30,30 +26,69 @@ export interface AppConfig extends IndexNameGetter {
...
+  autocompleteIndex: IndexVariation[]
+  lovIndex: IndexVariation[]
+  currencyIndex: IndexVariation[]
+  accessControlIndex: IndexVariation[]
+  hsoIndex: IndexVariation[]
+  raa: {
+    endpoint: string
+  }
+  pageSize: number
+  objectID: string
```
Comment (pavel-popov) [inline packages/core/src/configs/app-config.ts:47]: We probably don't need those for static configuration.

File: packages/core/src/configs/app-config.ts [hunk 2]
```diff
@@ -30,30 +26,69 @@ export interface AppConfig extends IndexNameGetter {
...
+}
+
+type IndeciesKeys =
```
Comment (pavel-popov) [inline packages/core/src/configs/app-config.ts]: IndexKeys

File: packages/core/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@findhotel/sapi-core",
-  "version": "1.0.0",
+  "version": "0.0.7",
```
Comment (jopdeklein) [inline packages/core/package.json:3]: 🅱️

File: packages/core/src/configs/app-config.ts [hunk 4]
```diff
@@ -30,30 +26,69 @@ export interface AppConfig extends IndexNameGetter {
 interface IndexVariation {
   name: string
   variationID?: string
+  objectID?: string
+}
+
+export interface SapiStaticConfig {
```
Comment (pavel-popov) [inline packages/core/src/configs/app-config.ts:32]: Simply `SapiConfig`. If client provides it, we use it, if not, then a client should provide AlgoliaClient and `objectID`, so we load the config from Algolia.

File: packages/core/src/configs/app-config.ts (hunk 2 above)
Comment (jopdeklein) [inline packages/core/src/configs/app-config.ts:50]:
```suggestion
type IndicesKeys =
```

File: packages/core/src/configs/app-config.ts [hunk 5]
```diff
@@ -30,30 +26,69 @@ export interface AppConfig extends IndexNameGetter {
...
+      config = results[0].hits[0] ?? {}
+    } else {
+      config = initWithConfig
+    }
 
     return {
       getIndexName: (name: IndexType) => {
-        const variations: IndexVariation[] = config[`${name}Index`]
+        const variations: IndexVariation[] =
+          config[`${name}Index` as IndeciesKeys]
```
Comment (jopdeklein) [inline packages/core/src/configs/app-config.ts:91]:
```suggestion
          config[`${name}Index` as IndicesKeys]
```

File: packages/core/src/sapi.ts [hunk 6]
```diff
@@ -55,6 +55,8 @@ export interface SapiClientOptions {
   deviceCategory: string
   /** A/B tests variation IDs */
   variationIds: Record<string, string>
+  /** SAPI static config to replace Algolia request */
```
Comment (jopdeklein) [inline packages/core/src/sapi.ts:58]:
```suggestion
  /** SAPI static config to override Algolia request */
```

**Review by jopdeklein — APPROVED** [review]

**Review by pavel-popov — APPROVED** [review]
LGTM, I left minor naming suggestions.

File: packages/core/src/configs/app-config.ts (hunk 1 above)
Comment (votsa) [reply packages/core/src/configs/app-config.ts:47]: Then description as well

**Pull request merged**